src/hook.c

```diff
@@ -757,3 +761,46 @@ MH_STATUS WINAPI MH_ApplyQueued(VOID)
         LeaveSpinLock();
     }
 }
+
+//-------------------------------------------------------------------------
+MH_STATUS WINAPI MH_CreateHookApi(LPCSTR pszTarget, LPVOID pDetour, LPVOID *ppOriginal)
+{
+    HMODULE hModule;
+    LPVOID  pTarget;
+    SIZE_T  copySize;
+    LPCSTR  psz = pszTarget;
+    LPCSTR  pszProcName = NULL;
+    CHAR    szModuleName[MAX_PATH];
+
+    if (pszTarget == NULL || pDetour == NULL)
+        return MH_ERROR_INVALID_ARGUMENTS;
+
+    // search the last '.' (replacement for strrchr())
+    while (*psz)
+        if (*psz++ == '.')
+            pszProcName = psz;
```

Comment:
Hi @RaMMicHaeL.
Check that ```while (*psz)``` loop doesn't have a ```break```. It looks an entire string for '.'.
Privately, I often hook the functions of ```ntdll.dll``` (or "undocumented" functions) so find this helper useful. As for Unicode file names, AFAIK, no system DLLs of Windows have Unicode file names.